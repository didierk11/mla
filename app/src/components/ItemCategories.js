import React from "react";

const ItemCategories = (props) => {
  const categories = props.categories;
  return (
    <div className="item-cat">
      <ol>
        {categories.map((item) => (
          <li key={item}>
            {item}
            <span className="chevron">{" > "}</span>
          </li>
        ))}
      </ol>
    </div>
  );
};

export default ItemCategories;
