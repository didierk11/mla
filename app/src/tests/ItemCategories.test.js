import React from "react";
import { render } from "react-dom";
import ItemCategories from "../components/ItemCategories";
import renderer from "react-test-renderer";

//testing data
const cat = ['testCategory1', 'testCategory2', 'testCategory3'];

describe("Snapshot test:", () => {
  it("ItemCategories render correctly", () => {
    const div = renderer.create(<ItemCategories categories={cat} />).toJSON();
    expect(div).toMatchSnapshot();
  });
});

describe("Smoke test:", () => {
  // ItemCategories
  it("ItemCategories render without crashing", () => {
    const div = document.createElement("div");
    render(<ItemCategories categories={cat} />, div);
  });
});