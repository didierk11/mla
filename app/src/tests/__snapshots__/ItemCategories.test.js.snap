// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test: ItemCategories render correctly 1`] = `
<div
  className="item-cat"
>
  <ol>
    <li>
      testCategory1
      <span
        className="chevron"
      >
         &gt; 
      </span>
    </li>
    <li>
      testCategory2
      <span
        className="chevron"
      >
         &gt; 
      </span>
    </li>
    <li>
      testCategory3
      <span
        className="chevron"
      >
         &gt; 
      </span>
    </li>
  </ol>
</div>
`;
